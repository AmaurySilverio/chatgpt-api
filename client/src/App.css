
#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

body {
  margin: 0;
  display: flex;
   /* place-items: center;  */
  min-width: 320px;
  min-height: 100vh;
}

body.light-mode {
  background-color: #fff;
  color: #333;
  transition: background-color 0.3s ease;
}
body.dark-mode {
  background-color: #1a1919;
  color: #999;
}

.message-container-history{
  width: 100%;
}
