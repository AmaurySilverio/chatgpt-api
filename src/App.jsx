// import { useState } from 'react'
// import './App.css'

function App() {

  return (
    <>
      <h2>CHAT GPT ASCEND</h2>
    </>
  )
}

export default App
