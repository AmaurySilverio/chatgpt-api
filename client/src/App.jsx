import React from "react";
import { Button } from "@nextui-org/react";
import useDarkMode from "use-dark-mode";
import './App.css'
import ThemeSwitcher from "./components/ThemeSwitcher";



export default function App() {

  const darkMode = useDarkMode(false, {
    classNameDark: "dark-mode",
    classNameLight: "light-mode",
    global: window, // Just pass this as a config option
  });

  return (
    <main className={`${darkMode ? darkMode.classNameDark : darkMode.classNameLight }`}>
      <ThemeSwitcher darkMode={darkMode} />
      <h2 className="text-3xl font-bold center underline">
        Chat GPT Ascend
      </h2>
      <Button radius="full" className="bg-gradient-to-tr from-pink-500 to-yellow-500 text-white shadow-lg">
       Test NextUI Component Button
      </Button>
    </main>
  );
}



