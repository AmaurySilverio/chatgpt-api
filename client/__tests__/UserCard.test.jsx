import React from 'react';
import { render } from 'vitest';
import UserCard from '../src/components/UserCard.jsx';
import { test } from 'vitest'; // 
import App from '../src/App.jsx';

test('renders user card with name and image', () => {
    window.HTMLElement.prototype.scrollIntoView = () => {};
    render(<App />);
 render(<UserCard></UserCard>)

    const user = {
        name: 'RC Student',
        imgUser: 'https://example.com/user.jpg'
    };

    const { getByText, getByAltText } = render(<UserCard name={user.name} imgUser={user.imgUser} />);

    expect(getByText(`Welcome, ${user.name}!`)).toBeInTheDocument(); // Use toBeInTheDocument() for text elements
    expect(getByAltText('RC Student')).toBeInTheDocument(); // Use toBeInTheDocument() for image elements
});
