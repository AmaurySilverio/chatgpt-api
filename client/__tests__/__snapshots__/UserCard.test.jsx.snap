// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders user card with name and image 1`] = `<UserCard />`;
